### DISCUSSION TITLE
Unable to have drag and drop working

### DISCUSSION THREAD

**Discussion by ptarczon** [abramenal/cypress-file-upload#341] [Q&A]
I would like to use `Drag-n-drop component`
```
cy.get('[data-cy="dropzone"]').attachFile('myfixture.json', { subjectType: 'drag-n-drop' });
```
to test uploading a file by imitating drag and drop.

I wrote a simple test:

```
it('drag and drop a pdf file', () => {
        cy.get('input[type="file"]')
            .attachFile(pdfFilePath, {subjectType: 'drag-n-drop'});
    })
```

Where `pdfFilePath` is the path to the pdf file and is declared like this:

```
const pdfFilePath = 'testFiles/pdf/Test_file.pdf'
```

And when running test it looks like it is executing the code:

![imageedit_0_3315224730](https://user-images.githubusercontent.com/88318881/140938623-b4283914-2462-459f-8ce0-3937ccf12a8b.png)

But in the application I don't see any file attached on the UI

Do you know what may be the reason and how to approach this?

Btw to be able to do HTML5 file input without encoding problems with upload of pdf file I had to add encoding:

```
cy.get('input[type="file"]')
        .attachFile({ filePath: pdfFilePath, encoding: 'base64'});
```

**Comment by krisodb** [comment]
When using drag-n-drop, check how the events are implemented in the website.
In my case I needed a "change" event. The normal drag, drop, etc events are not implemented at our side.
So it would become: cy.get('.upload__box').attachFile({filePath:'test_files/testDummy.pdf',encoding: "base64"},{subjectType:'drag-n-drop'}).trigger('change')

Note:  Would need to check if I have time, but for drag and drop I thought it's looking for an 'input[types=""file'] in underlying elements.  So I passed the element above as selector.
